cli/cmd/plugin/unmanaged-cluster/cluster/minikube.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 VMware Tanzu Community Edition contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cluster
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/vmware-tanzu/community-edition/cli/cmd/plugin/unmanaged-cluster/config"
+)
+
+var minikubeBinName = "minikube"
+
+type MinikubeClusterManager struct{}
+
+// Create
+func (mkcm MinikubeClusterManager) Create(c *config.UnmanagedClusterConfig) (*KubernetesCluster, error) {
+	if !mkcm.isMinikubeRunning() {
+		// TODO: Is minikube it the current context? If not, attempt to switch to it
+
+		args := []string{
+			"start",
+		}
+
+		cmd := exec.Command(minikubeBinName, args...)
```

Comment:
`minikube` has some pretty good logs that I believe would benefit persisting if the user needs to troubleshoot.

Could we switch this to:

```go
o, err := exec.Command(minikubeBinName, args...).Output()
```

^ there are a couple ways to do this, but most importantly writing the boostrap logs would be helpful.

Reply:
Just realized `minikube` has a log file related flag:

```
--log_file string                  If non-empty, use this log file
```

could be a simple solution here.

Reply:
Hmmmm. Not sure how to bring the bootstrapping log file down into the cluster provider.

Right now, the providers have _no_ notion of what logging the `tanzu` package is using. And the bootstrapping log file is associated with the `log.output` field on the `CMDLogger`.

I don't think we should introduce the precedence  that the providers know about any logging from a top level package.

We could keep a buffer and than bubble it back up to the `tanzu` package?

Reply:
Hmmmmmmm on second thought, maybe we do want to add a logger to the provider's API?

An example:

Let's say that a provider developer doesn't want to enable users to set the number of control-plane nodes. It'd be a pretty experience to have to dynamically check that in the `tanzu` package: the "orchestrator" would have to know about what the lower level "provider" expects and than log a warning on what it does.

Instead, if the provider had access to the logger, it could warn that it was going to ignore that configuration right in the provider code.